src/python/pants/engine/target.py

```diff
@@ -588,7 +588,12 @@ def register_plugin_field(cls, field: Type[Field]) -> UnionRule:
         return UnionRule(cls._plugin_field_cls, field)
 
     def validate(self) -> None:
-        """Validate the target, such as checking for mutually exclusive fields."""
+        """Validate the target, such as checking for mutually exclusive fields.
```

Comment:
I reached for this then slapped myself. In particular something like './pants metadata` or `./pants doc` or even `./pants package` or `./pants bdist_rpm` with support for `bdist_rpm`. Its just ~impossible to guess valid use cases for a target and the minute you seal in a validation here, you can't undo it without breaking unknown custom rule writers who now rely on it.